planet_rankings/planets.py

```diff
@@ -56,49 +56,46 @@
 # Here are the functions you need to implement to make things work.
 
 def get_user_selection(options):
-    """Present the user with options to choose from and return the selection.
-
-    Prints a "Number - Option" line for every option in options.
-    Asks the user to select an option by typing in its number.
-    Checks that the input represents a valid selection and returns it.
-    """
-
-    pass    # implement the function so that it mentions the docstring here
-
+    
+    #a list of options to choose from 
+    properties = (0,1,2,3,4,5)
+     
+    print(' 0 - Name')
+    print(' 1 - Diameter')
+    print(' 2 - Mass')
+    print(' 3 - Distance')
+    print(' 4 - Eccentricity')
+    print(' 5 - Moons')
+    
+    #ask and check for valid input and deal with the rest
+    while True: 
+        global x
+        try:
+            x = int(input('Auswahl: '))
+            if x in properties: 
+                return(x)
+            else:
+                print(' Please select one of the options given above.')
+                continue 
+        except ValueError:
+            print(' Please select one of the options given above.')
+            continue
 
 def sorted_on_attribute(data, sort_row):
-    """Return sorted (item, attribute) pairs from a larger data table.
-
-    Expects a data table in the form of a sequence, in which the first
-    element is a sequence of item names, and the following elements are
-    sequences of values describing one property of the items each.
-    sort_row indicates the index in the outer sequence that holds the
-    property to sort on. If sort_row is 0, the items are sorted by their
-    name and (item name, item name) tuples are returned.
-
-    Example:
-
-    >>> data = [
-        ('S.cerevisiae', 'C.elegans', 'H.sapiens', 'E.coli'),  # organisms
-        (12156677, 101169000, 3234830000, 4639221),    # haploid genome size
-        (16, 6, 23, 1)    # number of chromosomes in the haploid set
-        ]
-    >>> sorted_on_attribute(data, 2)
-    [('E.coli', 1), ('C.elegans', 6), ('S.cerevisiae', 16), ('H.sapiens', 23)]
     
-    """
+    #connect user selection with corresponding list 
+    #selection = list to zip and sort
+    
+    selection = solar_data[x] 
```

Comment:
viel besser ist hier:  `selection = data[sort_row]`
`solar_data[x]` funktioniert zwar auch, aber nur weil `solar_data` und `x` *globale* Variable im Hauptprogramm sind.
Sobald Du diese Funktion in einem anderen Kontext/Programm zu verwenden versuchst, gibt das Probleme.